lib/krikri/entity_behaviors/aggregation_entity_behavior.rb

```diff
@@ -0,0 +1,12 @@
+
+module Krikri
+  class AggregationEntityBehavior < Krikri::EntityBehavior
+    def generated_entities
+      @activity.generated_entity_uris.lazy.map do |uri|
+        agg = DPLA::MAP::Aggregation.new(uri)
+        agg.get   # slow?
```

Comment:
I think this is significantly less expensive than `OriginalRecord#load`, but still possibly something to be avoided here.  This call is already lazy, so the question is whether you'll ever want to access `#generated_entities` and get back unloaded objects (i.e. objects of the correct type, with the correct URI, but unpopulated).  I suspect the answer may be yes, but I don't see any use cases immediately.

Maybe the answer is something like:

```ruby
def entities(load = true)
  @activity.generated_entity_uris.lazy.map do |uri|
    agg = DPLA::MAP::Aggregation.new(uri)
    agg.get  if load
    agg
  end
end
```

Similarly for `OriginalRecord` you could do `load ? OriginalRecord.load(uri) : OriginalRecord.new(uri)`.